app/scripts/controllers/transactions/index.js

```diff
@@ -157,7 +159,7 @@ class TransactionController extends EventEmitter {
 
   async newUnapprovedTransaction (txParams, opts = {}) {
     log.debug(`MetaMaskController newUnapprovedTransaction ${JSON.stringify(txParams)}`)
-    const initialTxMeta = await this.addUnapprovedTransaction(txParams)
+    const initialTxMeta = await this.addUnapprovedTransaction(txParams, opts.origin)
     initialTxMeta.origin = opts.origin
```

Comment:
if this is already done in addUnapprovedTransaction we should remove this line

Reply:
We're not, but there's no reason we shouldn't. Speaking of which, at the end of `addUnapprovedTransaction` we do this (1):
```javascript
// save txMeta
this.txStateManager.updateTx(txMeta)
```

Then, immediately after in `newUnapprovedTransaction`, we do this (2):

```javascript
this.txStateManager.updateTx(initialTxMeta, '#newUnapprovedTransaction - adding the origin')
```

`addUnapprovedTransaction` is as far as I can tell only called in `newUnapprovedTransaction`. Can we remove the `updateTx` call in (2)?

Reply:
@rekmarks yeah it's change in this pr to do it in addUnapprovedTransaction

Reply:
oh whoops i acctualy miss read

Reply:
anyways if it's being passed in we might as well generate the txMeta with it

Reply:
so remove that line and put it in the initial txmeta opts where we call `this.txStateManager.generateTxMeta...` and add it as origin :D